mmpose/core/evaluation/mesh_eval.py

```diff
@@ -0,0 +1,68 @@
+# ------------------------------------------------------------------------------
+# Adapted from https://github.com/nkolot/SPIN
+# Original licence: Copyright (c) 2018 akanazawa, under the MIT License.
+# ------------------------------------------------------------------------------
+
+import numpy as np
+
+
+def compute_similarity_transform(source_points, target_points):
+    """Computes a similarity transform (sR, t) that takes a set of 3D points
+    source_points (N x 3) closest to a set of 3D points target_points, where R
+    is an 3x3 rotation matrix, t 3x1 translation, s scale. And return the
+    transformed 3D points source_points_hat (N x 3). i.e. solves the orthogonal
+    Procrutes problem.
+
+    Notes:
+        Points number: N
+
+    Args:
+        source_points (np.ndarray([N, 3])): Source point set.
+        target_points (np.ndarray([N, 3])): Target point set.
+
+    Returns:
+        source_points_hat (np.ndarray([N, 3])): Transformed source point set.
+    """
+
+    transposed = False
+    if source_points.shape[0] != 3 and source_points.shape[0] != 2:
```

Comment:
determine whether the data is transposed or not based on this heuristic is dangerous.
it is useless when there are only two or three points.

I suggest that we remove this guess and make it clear that the data shape should be `(N, 3)`